x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -423,7 +419,7 @@ public void testExtractorMultiEvalWithSameName() {
         var extract = as(project.child(), FieldExtractExec.class);
         assertThat(
             names(extract.attributesToExtract()),
-            contains("_meta_field", "emp_no", "first_name", "gender", "job.raw", "languages", "last_name", "salary")
+            contains("_meta_field", "emp_no", "first_name", "gender", "job", "job.raw", "languages", "last_name", "salary")
```

Comment:
Why are both job and job.raw extracted? Wouldn't job.raw be enough or does jaw leak somewhere?

Reply:
In the initial version of this PR, I [replaced all the occurrences](https://github.com/elastic/elasticsearch/pull/98528#discussion_r1304628316) in the plan with their exact subfield (including `EsRelation`), and as a result there was no need to load `job` anymore, but it made the planning a bit more convoluted, with the addition of some [aliases](https://github.com/elastic/elasticsearch/pull/98528#discussion_r1304629376) to keep the original names (otherwise the output for `from idx` would result in `job.raw` instead of `job`).
There could be a smarter way to do it and avoid fetching TEXT fields; in case we can further investigate and do it with a follow-up PR (at low level there are other optimizations, so we should not load from _source anyway)